commands/core/site_install.drush.inc

```diff
@@ -287,6 +298,21 @@ function drush_core_site_install($profile = NULL) {
       $form_options[$key] = $value;
     }
   }
+
+  // If the profile is not explicitly set, default to the 'minimal' for an issue-free config import.
+  if (empty($profile) && drush_get_option('config')) {
+    $profile = 'minimal';
```

Comment:
If the profile contains sub modules this is not going to work because the modules will not be found... actually what needs to happen here for correctness is the we should read the core.extension file in the incoming config and determine which is the install profile from the list of modules.